compiler/lib/flow.ml

```diff
@@ -321,6 +322,28 @@ let the_def_of info x =
         x
   | Pc c -> Some (Constant c)
 
+(* If [constant_identical a b = true], then the two values cannot be
+   distinguished, i.e., they are not different objects (and [caml_js_equals a b
+   = true]) and if both are floats, they are bitwise equal. *)
+let constant_identical a b =
```

Comment:
We need to adapt this function since boxed values can be distinguished

Reply:
So I guess the boxed values in Wasm but not in JS are: Float, NativeString and String? In short everything but Int?

Reply:
Right.

Reply:
Done